src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseCaseReport.java

```diff
@@ -0,0 +1,259 @@
+      return sample -> {
+        if (SAMPLE_CATEGORY_IDENTITY.equals(getAttribute(ATTR_CATEGORY, sample))) return false;
+        return timePointCode == getTimesReceived(sample.getName());
+      };
+    }
+  }
+
+  @Override
+  protected void collectData(PineryClient pinery) throws HttpResponseException, IOException {
+    List<SampleDto> allPreciseSamples =
+        pinery
+            .getSample()
+            .allFiltered(new SampleClient.SamplesFilter().withProjects(Lists.newArrayList("PRE")));
+    Map<String, SampleDto> allPreciseSamplesById = mapSamplesById(allPreciseSamples);
+    Map<String, List<SampleDto>> allPreciseSamplesByIdentity =
+        mapChildrenByIdentityId(allPreciseSamplesById);
+    List<String> tissueOriginOrder = Lists.newArrayList("Se", "Pl", "Ly");
+    List<String> tissueOriginOrderExtra = Lists.newArrayList("Se", "Pl", "Ly", "Us", "Up");
+    allPreciseIdentities =
+        allPreciseSamples
+            .stream()
+            .filter(s -> SAMPLE_CATEGORY_IDENTITY.equals(getAttribute(ATTR_CATEGORY, s)))
+            .collect(Collectors.toList());
+
+    table = new LinkedList<>();
+    List<SampleDto> potentialSamplesAtCurrentTime = null;
+    // One row per identity
+    for (SampleDto identity : allPreciseIdentities) {
+      List<String> row = new LinkedList<>();
+      row.add(identity.getName());
+
+      // For all regular time points
+      for (int i = 0; i < 5; i++) {
+        TimePoint currentTimePoint = TimePoint.values()[i];
+        potentialSamplesAtCurrentTime = allPreciseSamplesByIdentity.get(identity.getId());
+        if (potentialSamplesAtCurrentTime == null) {
+          row.addAll(
+              Lists.newArrayList(
+                  "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0", "0",
+                  "0", "0", "0", "0", "0", "0", "0", "0"));
+        } else {
+          List<SampleDto> samplesAtCurrentTime =
+              allPreciseSamplesByIdentity
+                  .get(identity.getId()) // Get children of this identity
+                  .stream()
+                  .filter(currentTimePoint.predicate()) // Get samples at this time point
+                  .collect(Collectors.toList());
+          // on 0th, 2th, 4th iterations use all the types, otherwise just 3
+          List<String> tissueOrigins;
+          if (i % 2 == 0) {
+            tissueOrigins = tissueOriginOrderExtra;
+          } else {
+            tissueOrigins = tissueOriginOrder;
+          }
+
+          // for all applicable tissue origins, get counts and add them to the row
+          for (String origin : tissueOrigins) {
+            row.add(
+                String.valueOf(
+                    samplesAtCurrentTime
+                        .stream()
+                        .filter(s -> origin.equals(getAttribute(ATTR_TISSUE_ORIGIN, s)))
+                        .count()));
+          }
+        }
+      }
+      if (potentialSamplesAtCurrentTime != null) {
```

Comment:
I think this means that it only looks for time points 6 and 7 if there are samples for time point 5? Might want to handle this differently in-case time point 5 gets skipped for some reason. Then `potentialSamplesAtCurrentTime` could be moved to a narrower scope as well.

Reply:
The result would be the same every loop, but i cleaned up the logic a little to save some wasted time

Reply:
Better not to make assumptions about future samples